### ISSUE TITLE
Text in card buttons doesn't wrap, making them difficult to use in FAQ bots

### ISSUE THREAD

**Issue by dividor** [Microsoft/BotFramework-Services#29]
We are using QnA maker to expose FAQs in our bot. For cases where we don't have a very high-scoring match, we present the user with the top 5 nearest questions as buttons, so they can choose their answer (if they confirm the answer, we use that to further train the QnA model).

The issue is, the text in buttons does not wrap, it gets truncated. This means that on mobile and even sometimes on Teams desktop, the user can be confused as they don't have the full question text for each question candidate.

We have tried other options, eg (i) using adaptive cards and selectAction to make the question candidates look like links. Trouble is, they don't look like buttons so users get confused; (ii) Using a carousel with a card for each question candidate. Again, in usability testing, users got confused.

Bottom line - in this scenario where we want to user to choose from a set of FAQs, users like choosing from a vertical stack of buttons, and so we really need to be able to wrap text and show the full question.

In fact, my question would be, why do we truncate text in buttons as the default? By truncating it makes the button useless as we lose information.

This issue is to ask if we can activate text wrapping for buttons, ideally in all cards, but our immediate need is for adaptive cards.

SDK: nodejs
Channels: Teams desktop+mobile; Webchat (though we can use CSS to wrap text in that channel)

**Comment by stevkan** [comment]
@dividor, are you using the v3 or v4 Node SDK?

**Comment by dividor** [comment]
@stevkan we are using v4 Node SDK. I attach the JSON

Hero card:

```
{
    "contentType": "application/vnd.microsoft.card.hero",
    "content": {
        "images": [],
        "buttons": [
            {
                "type": "imBack",
                "title": "How to merge platforms on system A",
                "value": "How to merge platforms on system A"
            },
            {
                "type": "imBack",
                "title": "How to add the City locations for platforms under clients in system A",
                "value": "How to add the City locations for platforms under clients in system A"
            },
            {
                "type": "imBack",
                "title": "How to change area leader in system A",
                "value": "How to change area leader in system A"
            },
            {
                "type": "imBack",
                "title": "How to add a library in system A",
                "value": "How to add a library in system A"
            },
            {
                "type": "imBack",
                "title": "How to enable notifications for a user in system A",
                "value": "How to enable notifications for a user in system A"
            },
            {
                "type": "postBack",
                "title": "None of the above",
                "value": {
                    "type": "intentRejection",
                    "confirmation": "No",
                    "modelType": "QnA",
                    "model": "ITKB-Internal-PROD-QnA",
                    "intent": "ITKB-Internal-PROD-QnA",
                    "userUtterance": "merge system A platforms",
                    "chosenQuestionId": "None of the above",
                    "chosenQuestion": "None of the above"
                }
            }
        ]
    }
}
```
This gives text truncation on one of the buttons ...

![btn](https://user-images.githubusercontent.com/8402586/50015374-6afcdf80-ff94-11e8-8b01-665436e814ad.PNG)

Here is the same for an adaptive card, which also truncates the button text:
```
{
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.0",
    "type": "AdaptiveCard",
    "speak": "",
    "body": [],
    "actions": [
        {
            "type": "Action.Submit",
            "title": "How to merge platforms on system A",
            "data": {
                "type": "faq_choice",
                "text": "How to merge platforms on system A"
            }
        },
        {
            "type": "Action.Submit",
            "title": "How to add the City locations for platforms under clients in system A",
            "data": {
                "type": "faq_choice",
                "text": "How to add the City locations for platforms under clients in system A"
            }
        },
        {
            "type": "Action.Submit",
            "title": "How to change area leader in system A",
            "data": {
                "type": "faq_choice",
                "text": "How to change area leader in system A"
            }
        },
        {
            "type": "Action.Submit",
            "title": "How to add a library in system A",
            "data": {
                "type": "faq_choice",
                "text": "How to add a library in system A"
            }
        },
        {
            "type": "Action.Submit",
            "title": "How to enable notifications for a user in system A",
            "data": {
                "type": "faq_choice",
                "text": "How to enable notifications for a user in system A"
            }
        },
        {
            "type": "Action.Submit",
            "title": "None of the above",
            "data": {
                "type": "intentRejection",
                "confirmation": "No",
                "modelType": "QnA",
                "model": "ITKB-Internal-PROD-QnA",
                "intent": "ITKB-Internal-PROD-QnA",
                "userUtterance": "merge system A platforms",
                "chosenQuestionId": "None of the above",
                "chosenQuestion": "None of the above"
            }
        }
    ]
}
```

**Comment by EricDahlvang** [comment]
This can be handled with some CSS workarounds as explained here https://stackoverflow.com/questions/48018290/using-botbuilder-sdks-prompt-choice-is-it-possible-to-have-a-custom-tooltip/48021311#48021311 , or by using a carousel of hero cards as explained here: https://github.com/Microsoft/BotFramework-WebChat/issues/616#issuecomment-320734680

If you feel there is something missing from the WebChat control, please file an issue here: https://github.com/Microsoft/BotFramework-WebChat/  Closing this issue, since it is not a bug or feature request.
